docs/topic-guides/commands-and-shells.txt

```diff
@@ -0,0 +1,135 @@
+persistent SSH server container in the Determined cluster and connect an
+interactive session to it, use ``det shell start``:
+
+.. code::
+
+   det shell start
+
+After starting a server with ``det shell start``, you can make another
+independent connection to the same server by running ``det shell open
+<UUID>``. The UUID can be obtained from the output of either the
+original ``det shell start`` command or ``det shell list``:
+
+.. code::
+
+   $ det shell list
+    Id                                   | Owner      | Description                  | State   | Exit Status
+   --------------------------------------+------------+------------------------------+---------+---------------
+    d75c3908-fb11-4fa5-852c-4c32ed30703b | determined | Shell (annually-alert-crane) | RUNNING | N/A
+   $ det shell open d75c3908-fb11-4fa5-852c-4c32ed30703b
+
+Optionally, you can provide extra options to pass to the SSH client when
+using ``det shell start`` or ``det shell open`` by including them after
+``--``. For example, this command will start a new shell and forward a
+port from the local machine to the container:
+
+.. code::
+
+   det shell start -- -L8080:localhost:8080
+
+In order to stop the SSH server container and free up cluster resources,
+run ``det shell kill <UUID>``.
+
+*********************
+ Context Directories
+*********************
+
+Commands and shells become much more powerful with the use of the ``-c
+<directory>`` option, which tells Determined to transfer files from a
+directory on the local machine (the "context directory") to the
+container. The contents of the context directory are placed into the
+directory ``/run/determined/workdir`` within the container before the
+command or shell starts running. ``/run/determined/workdir`` is also the
+initial working directory for commands, so they can easily access files
+from the context using relative paths.
+
+.. code::
+
+   $ mkdir context
+   $ echo 'print("hello world")' > context/run.py
+   $ det cmd run -c context python run.py
+
+The total size of the files in the context directory must be less than
+95 MB. Larger files, such as datasets, must be mounted into the
+container (see next section), downloaded after the container starts, or
+included in a :ref:`custom Docker image <custom-docker-images>`.
+
+************************
+ Advanced Configuration
+************************
+
+:ref:`Additional configuration settings
+<command-notebook-configuration>` for both commands and shells can be
+set using the ``--config`` and ``--config-file`` options. Commonly
+useful settings include:
+
+-  ``bind_mounts``: Specifies directories to be bind-mounted into the
+   container from the host machine. (Due to the structured values
+   required for this setting, it needs to be specified in a config
+   file.)
+
+-  ``resources.slots``: Specifies the number of slots the container will
+   have access to. (Distributed commands and shells are not supported;
+   all slots will be on one machine.)
```

Comment:
Say that if you request more slots than are available on a single machine, the command will never be scheduled?

Mention resource pools?

Reply:
> Say that if you request more slots than are available on a single machine, the command will never be scheduled?

Done!

> Mention resource pools?

I think I'll let this be batched in with the other doc updates for resource pools.